src/Raven.Server/Documents/Indexes/AbstractIndexCreateController.cs

```diff
@@ -223,6 +230,29 @@ private void ValidateAnalyzers(IndexDefinition definition)
             }
         }
     }
+    
+    private void ValidateConfiguration(IndexDefinition definition)
+    {
+        if (definition.Configuration == null)
+            return;
+        
+        foreach (var kvp in definition.Configuration)
+        {
+            if (_validConfigurationKeys.Contains(kvp.Key) == false)
+            {
+                throw new IndexCreationException($"Could not create index '{definition.Name}' because the configuration option key '{kvp.Key}' is not recognized");
+            }
+        }
+    }
+
+    private static HashSet<string> GetValidConfigurationKeys()
+    {
+        var allConfigurationEntries = RavenConfiguration.AllConfigurationEntries.Value;
+        
+        var validPerIndexConfigurationKeys = allConfigurationEntries.Where(configurationEntry => configurationEntry.Category == "Indexing").SelectMany(configurationEntry => configurationEntry.Keys).ToHashSet();
```

Comment:
The idea was to move this to a static field in `IndexingConfiguration`. Can we have it there?

Reply:
I moved the logic there